### PR TITLE
Invalidate the scope cache if a node is moved to a different path - fixes T7194

### PR DESCRIPTION
**Pull request by loganfsmyth**

### CODE REVIEW

File: packages/babel-traverse/src/scope/index.js [hunk 1]
```diff
@@ -23,17 +23,17 @@ let _crawlCallsCount = 0;
  * node itself containing all scopes it has been associated with.
```
Comment (hzoo) [inline packages/babel-traverse/src/scope/index.js]: We should remove/fixup the comment since the cache isn't on the node anymore

Comment (loganfsmyth) [inline packages/babel-traverse/src/scope/index.js]: The cache is still per-node, so I think this still applies.

Comment (hzoo) [inline packages/babel-traverse/src/scope/index.js]: Yeah I think so to, but its not on the `node itself` anymore right?

Comment (loganfsmyth) [inline packages/babel-traverse/src/scope/index.js]: Oh gotcha, you mean from the weakmap change. I suppose so.

**Comment by amasad** [comment]
LGTM

**Pull request merged**